src/ui/layout/tutorial.md

```diff
@@ -1,237 +1,386 @@
+  the top of the stack. Elements in the `children` array display
+  in array order on the screen from top to bottom.
+
+### Update the app to display the title section
+
+Add the `TitleSection` widget as the first element in the `children` array.
+This places it at the top of the screen.
+Pass the provided name and location to the `TitleSection` constructor.
+
+<?code-excerpt "{../base,step2}/lib/main.dart" from="children:" to="],"?>
+```diff
+--- ../base/lib/main.dart
++++ step2/lib/main.dart
+@@ -23 +19,6 @@
++            children: [
++              TitleSection(
++                name: 'Oeschinen Lake Campground',
++                location: 'Kandersteg, Switzerland',
++              ),
++            ],
 ```
 
 {{site.alert.tip}}
-  * When pasting code into your app, indentation can
-    become skewed. You can fix this in your Flutter editor
-    using the [automatic reformatting support][].
-  * For a faster development experience,
-    try Flutter's [hot reload][] feature.
-  * If you have problems, compare your code to [`lib/main.dart`][].
+
+* When pasting code into your app, indentation can become skewed.
+      To fix this in your Flutter editor, use [automatic reformatting support][].
+* To accelerate your development, try Flutter's [hot reload][] feature.
+* If you have problems, compare your code to [`lib/main.dart`][].
+
 {{site.alert.end}}
 
-## Step 3: Implement the button row
+## Add the Button section
+
+In this section, add the Buttons that will add functionality to your app.
 
 <?code-excerpt path-base="layout/lakes/step3"?>
 
-The button section contains 3 columns that use the same
-layout&mdash;an icon over a row of text.
-The columns in this row are evenly spaced,
-and the text and icons are painted with the primary color.
+The **Button** section contains three columns that use the same layout:
+an icon over a row of text.
+
+{% include docs/app-figure.md
+   image="ui/layout/layout-sketch-button-block-unlabeled.svg"
+   caption="The Button section as sketch and prototype UI" %}
+
+Plan to distribute these columns in one row so each takes the same
+amount of space. Paint all text and icons with the primary color.
 
-Since the code for building each column is almost identical,
-create a private helper method named `buildButtonColumn()`,
-which takes a color, an `Icon` and `Text`,
-and returns a column with its widgets painted in the given color.
+### Add the `ButtonSection` widget
 
-<?code-excerpt "lib/main.dart (_buildButtonColumn)" title?>
+Add the following code after the `TitleSection` widget to contain the code
+to build the row of buttons.
+
+<?code-excerpt "lib/main.dart (ButtonStart)" title?>
 ```dart
-class MyApp extends StatelessWidget {
-  const MyApp({super.key});
+class ButtonSection extends StatelessWidget {
+  const ButtonSection({super.key});
 
   @override
   Widget build(BuildContext context) {
-    // ···
+    final Color color = Theme.of(context).primaryColor;
+// ···
```

Comment:
It's a bit odd to see the plaster comment with no matching indentation. If you indent the nearby `enddocregion`, I think it should use that indentation.

Reply:
True, but I'd rather not re-run code-excerpts if it's not needed.

Reply:
That's fine. Do consider opening an issue to come back to it, or have someone else do so. Thanks!